src/Symfony/Component/Messenger/Bridge/Redis/Transport/Connection.php

```diff
@@ -521,6 +521,51 @@ public function cleanup(): void
         }
     }
 
+    public function getMessageCount(): int
+    {
+        $groups = $this->connection->xinfo('GROUPS', $this->stream);
+
+        if (false === $groups) {
+            throw new LogicException(sprintf('No group found for stream "%s".', $this->stream));
+        }
+
+        $lastDeliveredId = null;
+        foreach ($groups as $group) {
+            if ($group['name'] !== $this->group) {
+                continue;
+            }
+
+            // Use "lag" key provided by Redis 7.x. See https://redis.io/commands/xinfo-groups/#consumer-group-lag.
+            if (isset($group['lag'])) {
+                return $group['lag'];
+            }
+
+            $lastDeliveredId = $group['last-delivered-id'];
```

Comment:
Older php redis extension did not provide this information. See example: https://github.com/phpredis/phpredis/issues/1559

So we would maybe check if this is set and else also return a LogicException. Then the check for null === $lastDeliveredId should be obselete as I think redis always return a value here even if it is `0-0` for all versions supported this format.

Reply:
Sorry for the delay @alexander-schranz.

What about using `$lastDeliveredId = $group['last-delivered-id'] ?? '0-0';` to have a default value if we cannot retrieve this information from the group. We could also keep the check for null value in the case where no group was found in the loop.

WDYT ?

Reply:
Sorry for the late reply, I missed your previous comments. 
I agree that a proper exception would probably be better, but for now it seems the existing implementations are just returning 0 in case they can't get the info, at least: https://github.com/symfony/symfony/blob/a05702965e9fc0de7ce77d1641e772e2fbfe9911/src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php#L314
But if there is really an issue about the setup (i.e. a developer error than can be fixed) then throwing a messenger's LogicException sounds good as long as it does not break existing setups.

Reply:
👍 We can improve that later if needed

Reply:
OK. I have updated the code to return 0 if an error occurs. No exception will be thrown.